unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Factories/ServiceLocatorFactory/ServiceLocatorFactory.cs

```diff
@@ -37,13 +37,20 @@ public static ServiceLocator CreateDefault()
             result.Register<IParcelScenesCleaner>(() => new ParcelScenesCleaner());
             result.Register<IClipboard>(Clipboard.Create);
             result.Register<IPhysicsSyncController>(() => new PhysicsSyncController());
-            result.Register<IWebRequestController>(WebRequestController.Create);
+            result.Register<IRPC>(() => new RPC());
+            result.Register<IWebRequestController>(() => new WebRequestController(
+                new GetWebRequestFactory(),
+                new WebRequestAssetBundleFactory(),
+                new WebRequestTextureFactory(),
+                new WebRequestAudioFactory(),
+                new PostWebRequestFactory(),
+                RPCSignRequest.CreateSharedInstance(result.Get<IRPC>())
```

Comment:
Introduce a local var for `IPRC`: it's a bad practice to resolve dependencies during registration. Or you can use `Service<>` in the `RPCSignRequest` itself